pandas/core/internals/blocks.py

```diff
@@ -1741,9 +1741,14 @@ def fillna(
                 downcast=downcast,
                 using_cow=using_cow,
             )
-        new_values = self.values.fillna(value=value, method=None, limit=limit)
-        nb = self.make_block_same_class(new_values)
-        return nb._maybe_downcast([nb], downcast)
+        if using_cow and self._can_hold_na and not self.values.isna().any():
```

Comment:
```suggestion
        if using_cow and self._can_hold_na and not self.values._hasna:
```

(for some EAs that can be optimized)

Reply:
updated